services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedTenantService.java

```diff
@@ -228,23 +230,25 @@ public void get(final String tenantId, final SpanContext spanContext, final Hand
     }
 
     @Override
-    public void get(final X500Principal subjectDn, final SpanContext spanContext,
+    public void get(final X500Principal subjectDn, final Span span,
             final Handler<AsyncResult<TenantResult<JsonObject>>> resultHandler) {
 
         Objects.requireNonNull(subjectDn);
         Objects.requireNonNull(resultHandler);
 
-        resultHandler.handle(Future.succeededFuture(getForCertificateAuthority(subjectDn)));
+        resultHandler.handle(Future.succeededFuture(getForCertificateAuthority(subjectDn, span)));
     }
 
-    private TenantResult<JsonObject> getForCertificateAuthority(final X500Principal subjectDn) {
+    private TenantResult<JsonObject> getForCertificateAuthority(final X500Principal subjectDn, final Span span) {
 
         if (subjectDn == null) {
+            TracingHelper.logError(span, "missing subject DN");
             return TenantResult.from(HttpURLConnection.HTTP_BAD_REQUEST);
         } else {
             final TenantObject tenant = getByCa(subjectDn);
 
             if (tenant == null) {
+                TracingHelper.logError(span, "tenant not found");
```

Comment:
I think it is clear from the status code that the tenant has not been found. IMHO the log message would be more helpful if it would explain the reason, in this case something like `no tenant found for subject DN`.

Reply:
I've changed the log message accordingly.